dpctl/tensor/_dlpack.pyx

```diff
@@ -718,6 +718,11 @@ class _numpy_array_interface_wrapper:
         self._memory_owner = memory_owner
 
 
+cdef bint _is_kdlcpu_device(DLDevice *dev):
+    "Check if DLTensor.DLDevice denotes (kDLCPU, 0)"
+    return (dev.device_type == kDLCPU) and (dev.device_id == 0)
```

Comment:
From C perspective, this reads odd. In C, this should translate to `dev->device_type` and `dev->device_id`. To be able to use struct member access like this, my reaction would be to write `dev[0].device_type` and `dev[0].device_id`. 

But perhaps Cython handles this gracefully. I would check generated `cxx` file to confirm that correct C++ code is being generated.

Reply:
It looks correct in `_dlpack.cxx`

```
  __Pyx_TraceLine(723,0,__PYX_ERR(0, 723, __pyx_L1_error))
  __pyx_t_2 = (__pyx_v_dev->device_type == kDLCPU);
  if (__pyx_t_2) {
  } else {
    __pyx_t_1 = __pyx_t_2;
    goto __pyx_L3_bool_binop_done;
  }
  __pyx_t_2 = (__pyx_v_dev->device_id == 0);
  __pyx_t_1 = __pyx_t_2;
  __pyx_L3_bool_binop_done:;
  __pyx_r = __pyx_t_1;
  goto __pyx_L0;
```

But it wouldn't take much to go ahead and change it anyway.